src/main/java/com/google/devtools/build/lib/skylarkbuildapi/platform/ConstraintCollectionApi.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.skylarkbuildapi.platform;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.skylarkinterface.Param;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkCallable;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModule;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModuleCategory;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkValue;
+import com.google.devtools.build.lib.syntax.SkylarkIndexable;
+import javax.annotation.Nullable;
+
+/** Info object representing data about a specific platform. */
+@SkylarkModule(
+    name = "ConstraintCollection",
```

Comment:
I can send a doc, although I was hoping to keep this simple enough.

ConstraintCollectionApi implements SkylarkIndexable, and the ConstraintCollection implementation handles this properly, so practically, Starlark users can access it as if it were a dict:
constraint_value = platform_info.constraints[setting]

The reason to add a new class is because we need to support default values for constraint settings, so the ConstrainCollection itself needs some intelligence.

Reply:
Final decision was to declare platform APIs as experimental and not subject to support.